### PR TITLE
Stabilize server and improve logging

### PR DESCRIPTION
**Pull request by robertschaedler3**
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/platform/orchestrator/src/MpiServer.cpp [hunk 1]
```diff
@@ -594,16 +594,24 @@ static bool ReadRequest(int connfd, Request& request)
         {
             if (contentLength != (bytesRead = read(connfd, buffer, contentLength)))
             {
-                OsConfigLogError(PlatformLog::Get(), "Failed to read complete HTTP body: Content-Length %d, bytes read %d", contentLength, (int)bytesRead);
+                if (IsFullLoggingEnabled())
+                {
+                    OsConfigLogError(PlatformLog::Get(), "Failed to read complete HTTP body: '%s' Content-Length %d, bytes read %d, body ' %.*s'", uri, contentLength, static_cast<int>(bytesRead), static_cast<int>(bytesRead), buffer);
```
Comment (MariusNi) [inline src/platform/orchestrator/src/MpiServer.cpp:599]: >"Failed to read complete HTTP body: '%s' Content-Length %d, bytes read %d, body ' %.*s'",  [](http://example.com/codeflow?start=57&length=90)

Why not just:

Same as before +

"Failed to read complete HTTP body for '%s'...."

Comment (MariusNi) [reply src/platform/orchestrator/src/MpiServer.cpp:599]: Or even w/o ''

"%s: failed to read complete HTTP body...."

where %s is the uri

File: src/platform/orchestrator/src/MpiServer.cpp [hunk 2]
```diff
@@ -115,28 +101,97 @@ std::string CreateGUID()
...
+            }
+        }
+        else
+        {
+            OsConfigLogError(PlatformLog::Get(), "Failed to bind socket: '%s'", g_mpiSocket);
+        }
+    }
+    else
+    {
+        OsConfigLogError(PlatformLog::Get(), "Failed to create socket: '%s'", g_mpiSocket);
```
Comment (MariusNi) [inline src/platform/orchestrator/src/MpiServer.cpp:174]: nit. Can get rid of : and ''. Just say "Blah socket %s"

File: src/platform/orchestrator/src/MpiServer.cpp [hunk 3]
```diff
@@ -115,28 +101,97 @@ std::string CreateGUID()
...
+            result = "OK";
+            break;
+        case StatusCode::BAD_REQUEST:
+            result = "BAD_REQUEST";
+            break;
+        case StatusCode::NOT_FOUND:
+            result = "NOT_FOUND";
+            break;
+        default:
+            result = "INTERNAL_SERVER_ERROR";
```
Comment (MariusNi) [inline src/platform/orchestrator/src/MpiServer.cpp:119]: These can change to "ok", "bad request", "not found", etc. If used for logging.

File: src/platform/orchestrator/src/MpiServer.cpp [hunk 4]
```diff
@@ -471,203 +542,115 @@ static void MpiGetReportedRequest(const Request& request, Response& response)
...
+        return;
     }
-    else if (0 >= (contentLength = ReadHttpContentLengthFromSocket(connfd, PlatformLog::Get())))
+
+    if (0 >= (contentLength = ReadHttpContentLengthFromSocket(connfd, PlatformLog::Get())))
     {
-        OsConfigLogError(PlatformLog::Get(), "Failed to read HTTP Content-Length");
-        success = false;
+        OsConfigLogError(PlatformLog::Get(), "Failed to read HTTP Content-Length for '%s'", uri);
+        return;
```
Comment (MariusNi) [inline src/platform/orchestrator/src/MpiServer.cpp:614]: Change to:

"%s: failed to read Content-Length"

Comment (MariusNi) [reply src/platform/orchestrator/src/MpiServer.cpp:614]: Similar in the other cases

**Comment by MariusNi** [comment]
> 

Blank line 700 - remove

---
Refers to: src/platform/orchestrator/src/MpiServer.cpp:700 in 823844b. [](commit_id = 823844ba7a7ca9d2dfb8bf480ac27ca29e86acc8, deletion_comment = False)

File: src/platform/orchestrator/src/MpiServer.cpp (hunk 3 above)
Comment (robertschaedler3) [reply src/platform/orchestrator/src/MpiServer.cpp:119]: Not using these for logging, only for creating the response header.

**Review by MariusNi — DISMISSED** [review]
:shipit:

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**